gammapy/datasets/spectrum.py

```diff
@@ -325,15 +327,28 @@ def read(cls, filename, format="ogip", checksum=False, name=None, **kwargs):
         """
         from .io import OGIPDatasetReader
 
-        if format == "gadf":
-            return super().read(
-                filename, format="gadf", checksum=checksum, name=name, **kwargs
-            )
+        if format is None:
+            with fits.open(filename) as hdulist:
+                # Check if the file is empty
+                if len(hdulist) == 0:
+                    raise ValueError(f"File {filename} is empty")
+                # Check for extensions in OGIP format
+                if "SPECTRUM" in hdulist and "OGIP" in hdulist[1].header["HDUCLASS"]:
+                    format = "ogip"
+                # Check for extensions in GADF format
+                elif ("COUNTS" in hdulist and "EXPOSURE" in hdulist and "EDISP" in hdulist):
+                    format = "gadf"
```

Comment:
given the limited amount of info currently serialized this will be difficult to identify in the future. e.g. so far we have not serialized the `HDUCLASS` in the so-called gadf format...
So you could check that there is a `COUNTS_BAND` extension.